vpn/client.go

```diff
@@ -0,0 +1,154 @@
+package vpn
+
+import (
+	"context"
+	"net/http"
+	"net/netip"
+
+	"github.com/tailscale/wireguard-go/tun"
+	"golang.org/x/xerrors"
+	"nhooyr.io/websocket"
+	"tailscale.com/net/dns"
+	"tailscale.com/wgengine/router"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/codersdk/workspacesdk"
+	"github.com/coder/coder/v2/tailnet"
+	"github.com/coder/coder/v2/tailnet/proto"
+	"github.com/coder/quartz"
+)
+
+type Client struct {
+	sdk *codersdk.Client
+}
+
+// NewClient creates a new VPN client.
+func NewClient(c *codersdk.Client) *Client {
+	return &Client{
+		sdk: c,
+	}
+}
+
+type DialOptions struct {
```

Comment:
I find this name confusing.  "DialOptions" sounds like it should be settings related to dialing the control plane, but many of these things are unrelated to dialing.

I can see why you named it this way, as in, it's the options passed to the `Dial()` function.  Maybe that function should be renamed, because it does way more than just dial the control plane: it instantiates the `tailnet` data plane with all its options as well.  `NewConn()` / `ConnOptions` ?